src/combinatorics/simple_graph/basic.lean

```diff
@@ -576,4 +504,49 @@ end
 
 end complement
 
+section maps
+
+/-- A graph homomorphism is a map on vertex sets that respects adjacency relations -/
+abbreviation homomorphism (G G' : simple_graph V) := rel_hom (G.adj) (G'.adj)
```

Comment:
I don't think it's a good idea to insist that graph homomorphisms are only defined on graphs with the same vertex type.

Reply:
OK I will not require the same vertex type.